src/components/bookDetail/ReviewBox.tsx

```diff
@@ -0,0 +1,272 @@
+
+  const addReview = useMutation({
+    mutationFn: async () => {
+      if (!user) return;
+      if (!bookIsbn) return;
+      if (!reviewText) return;
+
+      return await api.updateReview({
+        userId: user.id,
+        nickname: user.nickname,
+        bookIsbn: bookIsbn,
+        reviewId: generateId(),
+        content: reviewText,
+      });
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['reviews', bookIsbn] });
+      setReviewText('');
+      alert('리뷰 작성 완료');
+    },
+    onError: () => {
+      alert('추가 실패!');
+    },
+  });
+
+  const updateReview = useMutation({
+    mutationFn: async (changedReview: Review) => {
+      if (!user) return;
+      if (!bookIsbn) return;
+      if (!updateText) return;
+
+      return await api.updateReview({
+        ...changedReview,
+        content: updateText,
+      });
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['reviews', bookIsbn] });
+      setUpdateTarget(undefined);
+      setUpdateText('');
+      alert('리뷰 수정 완료');
+    },
+    onError: () => {
+      alert('수정 실패!');
+    },
+  });
+
+  const { data: reviews } = useQuery({
+    queryKey: ['reviews', bookIsbn],
+    queryFn: async () => {
+      if (!bookIsbn) return [];
+
+      const result = await api.getReviewsByBookIsbn(bookIsbn);
+      return result;
+    },
+    enabled: !!bookIsbn,
+    retry: false,
+  });
+
+  const deleteReview = useMutation({
+    mutationFn: (reviewId: string) => api.deleteMemoByReviewId(reviewId),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['reviews', bookIsbn] });
+      alert('리뷰 삭제');
+    },
+    onError: () => {
+      alert('제거 실패!');
+    },
+  });
+
+  const startReviewUpdateHandle = (target: Review) => {
+    setUpdateTarget(target);
+    setUpdateText(target.content);
+  };
+
+  return (
+    <>
+      <Wrapper>
+        <Flex alignItems={'baseline'} gap={'10px'}>
+          <DescriptionTitle>리뷰</DescriptionTitle>
+          <ReviewSubText>이 책을 평가해주세요!</ReviewSubText>
+        </Flex>
+        <HorizontalLine margin="0 0 16px"></HorizontalLine>
+        <Flex flexDirection={'column'} alignItems={'flex-end'}>
+          <ReviewTextarea
+            placeholder="리뷰 작성 시 광고 및 욕설, 비속어나 타인을 비방하는 문구를 사용하시면 삭제될 수 있습니다"
+            maxLength={100}
+            disabled={!isAuthenticated}
+            value={reviewText}
+            onChange={(e) => {
+              setReviewText(e.target.value);
+            }}
+          />
+          <Button
+            disabled={!isAuthenticated}
+            onClick={() => {
+              addReview.mutate();
+            }}
+          >
+            리뷰 남기기
+          </Button>
+        </Flex>
+        <ReviewListWrapper>
+          <DescriptionTitle>
+            총 <strong>{reviews ? reviews.length : 0}</strong> 개 리뷰
+          </DescriptionTitle>
+          <HorizontalLine margin="0 0 16px"></HorizontalLine>
+
+          {!!reviews &&
```

Comment:
```suggestion
          {reviews.length > 0 &&
```

배열은 `!![] === true` 이기 때문에 length로 체크해줘야해요.